### PR TITLE
CI: do not use source cache, but artifacts instead

### PR DESCRIPTION
**Pull request by v01d**
## Summary

This removes use of source cache on CI and uses artifacts instead. The cache sometimes fails and fails the entire build.

## Impact

CI

## Testing

Test via CI of this PR itself

### CODE REVIEW

**Comment by btashton** [comment]
@v01d can you just remove the use of the cache for the source.  With the new github cache infra the cache is never ready in time and it make this more complex.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -121,7 +121,7 @@ jobs:
         uses: actions/cache@v2
         with:
           path: sources
-          key: build-sources-${{ github.run_id }}
+          key: build-sources-${{ github.run_id }}-${{ github.run_number }}
```
Comment (xiaoxiang781216) [inline .github/workflows/build.yml:124]: Should we modify the same file in apps and testing git?

Comment (btashton) [reply .github/workflows/build.yml:124]: We need to just drop the cache.  We always miss no that GitHub has made cache faster.

**Comment by v01d** [comment]
So I should just leave the source passing between jobs via artifacts as it happens on cache miss?

**Comment by btashton** [comment]
> So I should just leave the source passing between jobs via artifacts as it happens on cache miss?

Yeah, we can also move from `actions/upload-artifact@v1` to `actions/upload-artifact@v2`  (and the download) and drop having the extra create source bundle step where the tar.gz is created.  That will improve the performance on that step as well.
V2 supports directory and not just file uploads https://github.com/actions/upload-artifact#upload-an-entire-directory

**Comment by v01d** [comment]
Ok, I can do that. Just checked the v2 and this is a bit concerning if the directory is used instead of tar: https://github.com/actions/upload-artifact#case-insensitive-uploads
If you don't think it is a problem I can follow your suggestion.

**Comment by v01d** [comment]
I just updated the PR to reflect the new way

**Comment by btashton** [comment]
> Ok, I can do that. Just checked the v2 and this is a bit concerning if the directory is used instead of tar: https://github.com/actions/upload-artifact#case-insensitive-uploads
> If you don't think it is a problem I can follow your suggestion.

Good catch.  Mostly on the permissions bit limitation.  I think it makes sense to just do the tar not tar.gz as they show, that way we do not compress twice.

**Comment by v01d** [comment]
Ok, I just pushed without the tar, I can bring it back.

**Comment by btashton** [comment]
Looks like uploading a directory is a bad idea, they seem to be streaming each file.  So yeah lets move back to tar as it was before.

**Comment by v01d** [comment]
Yup, I added the tar step and moved back to v1 as it failed with a permission problem.

**Comment by v01d** [comment]
Ok, should be good to merge.

**Pull request merged**